src/internal/connector/exchange/service_iterators_test.go

```diff
@@ -0,0 +1,648 @@
+	var (
+		userID   = "user_id"
+		tenantID = tester.M365TenantID(suite.T())
+		cat      = path.EmailCategory // doesn't matter which one we use,
+		qp       = graph.QueryParams{
+			Category:      cat,
+			ResourceOwner: userID,
+			Credentials:   suite.creds,
+		}
+		statusUpdater = func(*support.ConnectorOperationStatus) {}
+		allScope      = selectors.NewExchangeBackup(nil).MailFolders(selectors.Any(), selectors.Any())[0]
+		commonResults = mockGetterResults{
+			added:    []string{"added"},
+			newDelta: api.DeltaUpdate{URL: "new_delta_url"},
+		}
+		expiredResults = mockGetterResults{
+			added: []string{"added"},
+			newDelta: api.DeltaUpdate{
+				URL:   "new_delta_url",
+				Reset: true,
+			},
+		}
+	)
+
+	prevPath := func(t *testing.T, at ...string) path.Path {
+		p, err := path.Builder{}.
+			Append(at...).
+			ToDataLayerExchangePathForCategory(tenantID, userID, cat, false)
+		require.NoError(t, err)
+
+		return p
+	}
+
+	type endState struct {
+		state      data.CollectionState
+		doNotMerge bool
+	}
+
+	table := []struct {
+		name     string
+		getter   mockGetter
+		resolver graph.ContainerResolver
+		dps      DeltaPaths
+		expect   map[string]endState
+	}{
+		{
+			name: "new container",
+			getter: map[string]mockGetterResults{
+				"1": commonResults,
+			},
+			resolver: newMockResolver(mockResolvedContainer{
+				id:          "1",
+				displayName: "new",
+				p:           path.Builder{}.Append("1", "new"),
+			}),
+			dps: DeltaPaths{},
+			expect: map[string]endState{
+				"1": {data.NewState, false},
+			},
+		},
+		{
+			name: "not moved container",
+			getter: map[string]mockGetterResults{
+				"1": commonResults,
+			},
+			resolver: newMockResolver(mockResolvedContainer{
+				id:          "1",
+				displayName: "not_moved",
+				p:           path.Builder{}.Append("1", "not_moved"),
+			}),
+			dps: DeltaPaths{
+				"1": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "1", "not_moved").String(),
+				},
+			},
+			expect: map[string]endState{
+				"1": {data.NotMovedState, false},
+			},
+		},
+		{
+			name: "moved container",
+			getter: map[string]mockGetterResults{
+				"1": commonResults,
+			},
+			resolver: newMockResolver(mockResolvedContainer{
+				id:          "1",
+				displayName: "moved",
+				p:           path.Builder{}.Append("1", "moved"),
+			}),
+			dps: DeltaPaths{
+				"1": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "1", "prev").String(),
+				},
+			},
+			expect: map[string]endState{
+				"1": {data.MovedState, false},
+			},
+		},
+		{
+			name:     "deleted container",
+			getter:   map[string]mockGetterResults{},
+			resolver: newMockResolver(),
+			dps: DeltaPaths{
+				"1": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "1", "deleted").String(),
+				},
+			},
+			expect: map[string]endState{
+				"1": {data.DeletedState, false},
+			},
+		},
+		{
+			name: "one deleted, one new",
```

Comment:
I know there's already a lot of test cases here, but I think two more (very similar) ones would be good to have:
* delete folder a, make new folder with same path
* move folder a, make new folder with same path as old location of a

Reply:
good ideas!